HACKING.md

```diff
@@ -0,0 +1,160 @@
+# Hacking on snapd
+
+Hacking on snapd is fun and straightfoward. The code is extensively
+unit tested and we use the [spread](https://github.com/snapcore/spread)
+integration test framework for the integration/system level tests.
+
+## Development
+
+### Setting up a GOPATH
+
+When working with the source of Go programs, you should define a path within
+your home directory (or other workspace) which will be your `GOPATH`. `GOPATH`
+is similar to Java's `CLASSPATH` or Python's `~/.local`. `GOPATH` is documented
+[http://golang.org/pkg/go/build/](online) and inside the go tool itself
+
+    go help gopath
+
+Various conventions exist for naming the location of your `GOPATH`, but it
+should exist, and be writable by you. For example
+
+    export GOPATH=${HOME}/work
+    mkdir $GOPATH
+
+will define and create `$HOME/work` as your local `GOPATH`. The `go` tool
+itself will create three subdirectories inside your `GOPATH` when required;
+`src`, `pkg` and `bin`, which hold the source of Go programs, compiled packages
+and compiled binaries, respectively.
+
+Setting `GOPATH` correctly is critical when developing Go programs. Set and
+export it as part of your login script.
+
+Add `$GOPATH/bin` to your `PATH`, so you can run the go programs you install:
+
+    PATH="$PATH:$GOPATH/bin"
+
+### Getting the snappy sources
+
+The easiest way to get the source for `snappy` is to use the `go get` command.
+
+    go get -d -v github.com/snapcore/snapd/...
+
+This command will checkout the source of `snappy` and inspect it for any unmet
+Go package dependencies, downloading those as well. `go get` will also build
+and install `snappy` and its dependencies. To checkout without installing, use
```

Comment:
snapd.

This paragraph seems to assume the command we just told the user to do was without `-d`; I think it needs reversing (e.g “to also build and install `snapd` itself into `${GOPATH}/bin`, omit the `-d` flag”)